domain/consensus/processes/coinbasemanager/block_reward_switch.go

```diff
@@ -40,10 +37,28 @@ func (c *coinbaseManager) isBlockRewardFixed(stagingArea *model.StagingArea) (bo
 		blueScoreDifference := new(big.Int).SetUint64(highPruningPointHeader.BlueScore() - lowPruningPointHeader.BlueScore())
 		estimatedAverageHashRate := new(big.Int).Div(blueWorkDifference, blueScoreDifference)
 		if estimatedAverageHashRate.Cmp(c.fixedSubsidySwitchHashRateDifference) >= 0 {
-			c.hasBlockRewardSwitchedToFixed = true
 			return true, nil
 		}
 	}
 
 	return false, nil
 }
+
+func (c *coinbaseManager) findPruningPointIndex(stagingArea *model.StagingArea, pruningPointHash *externalapi.DomainHash) (uint64, error) {
+	currentPruningPointHash, err := c.pruningStore.PruningPoint(c.databaseContext, stagingArea)
+	if err != nil {
+		return 0, err
+	}
+	currentPruningPointIndex, err := c.pruningStore.CurrentPruningPointIndex(c.databaseContext, stagingArea)
+	if err != nil {
+		return 0, err
+	}
+	for !currentPruningPointHash.Equal(pruningPointHash) && currentPruningPointIndex > 0 {
```

Comment:
Why `currentPruningPointIndex > 0`? Don't we want to return an error if we get to zero and still don't find it?